packages/editor/src/components/post-author/index.js

```diff
@@ -1,49 +1,165 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { sprintf, __, _n } from '@wordpress/i18n';
 import { withInstanceId, compose } from '@wordpress/compose';
 import { Component } from '@wordpress/element';
 import { withSelect, withDispatch } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
+/**
+ * External dependencies
+ */
+import Autocomplete from 'accessible-autocomplete/react';
+
+import { debounce } from 'lodash';
+
 export class PostAuthor extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.setAuthorId = this.setAuthorId.bind( this );
+		this.suggestAuthor = this.suggestAuthor.bind( this );
+		this.getCurrentAuthor = this.getCurrentAuthor.bind( this );
+		this.resolveResults = this.resolveResults.bind( this );
+		this.state = {
+			postAuthor: false,
+		};
+		this.searchCache = [];
+		this.requestResults = debounce( ( query, populateResults ) => {
+			const payload = '?search=' + encodeURIComponent( query );
+			apiFetch( { path: `/wp/v2/users${ payload }` } ).then( ( results ) => {
+				populateResults( this.resolveResults( results ) );
+				this.searchCache[ query ] = results;
+			} );
+		}, 300 );
+		this.requestResults = this.requestResults.bind( this );
+	}
+
+	componentDidMount() {
+		const { postAuthorId, authors, postAuthor } = this.props;
+
+		// If the postAuthor is provided, use it directly.
+		if ( postAuthor ) {
+			this.setState( { postAuthor } );
```

Comment:
Duplicating state from props is an anti-pattern, as it muddies the source of truth. Why can't you just use `this.props.postAuthor` directly wherever it is referenced by state?

**Edit:** Okay, I see in the following logic that a fallback behavior occurs when the prop is not provided. The issue of duplicating the prop still remains, as it becomes a syncing concern. The prop can change and you don't appear to be reconciling it. I think this ought to be refactored to a getter function which returns either the prop value or the fallback logic ([see also from a cached copy of an earlier version of React documentation](https://zhenyong.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html)).

Reply:
I can switch to a getter, that might clean up the logic. I think this handles new posts which have no author (and instead use the current user)

Reply:
added a getter for this in https://github.com/WordPress/gutenberg/pull/7385/commits/fcaf60d11cc880bb058d95f0f998544cb9fec303